### DISCUSSION TITLE
Are the scripts necessary or can Vite do everything?

### DISCUSSION THREAD

**Discussion by billyzkid** [Deluze/electron-vue-template#48] [General]
I'm new to Electron, Vue, and Vite development, but have been doing a little reading and looking at some of the various electron-vue-vite templates out there.

I like this template because unlike many others, it doesn't have a lot of magic going on and seems to have minimum configuration. It's very easy to understand and still all the main features including HMR, etc.

One thing I'm not sure about though are the build and dev-server scripts. I see that they are using the Vite APIs to build the `renderer` files, but why do both the scripts needs to do manual `tsc` compiling of the `main` source files, starting and restarting Electron when files change, etc.? Couldn't that also be the job of Vite? Do we even need the scripts?

**Comment by Deluze** [comment]
Hey :)

We need to start Vite's dev server by code because Electron should know on what port the dev server is running, hardcoding the port according to `vite.config.js` would make it impossible to run multiple dev instances. AFAIK there's no easier way to go about this. HMR works out of the box as Vite's server is able to transpile & serve the TypeScript files as JavaScript, but running a Vite server for Electron isn't possible as it's running in a JavaScript runtime environment (NodeJS...)

The `tsc` script is currently only used to build the TypeScript files of Electron, however, I think Vite should be able to build these too just fine 🤔 so you could be right.. 

I was planning to do some digging on Electron Forge instead of Electron Builder, as it's maintained by Electron themselves and has Vite built in, I think it would eliminate most, if not all of these scripts.

Unsure if any of these statements are still relevant, I open-sourced this project as I used it a lot for my own a while back, and it just worked ^^, it's getting a lot of traction so I don't mind spending some time in the near future to modernise this template..
